box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java

```diff
@@ -0,0 +1,275 @@
+     * @return the transformed data
+     */
+    private DataWrapper<BoxCollaborationItem> createFetchRoleItemData(BoxResponse<BoxCollaborationItem> response) {
+        final DataWrapper<BoxCollaborationItem> data = new DataWrapper<BoxCollaborationItem>();
+        if (response.isSuccess()) {
+            BoxCollaborationItem collaborationItem = response.getResult();
+            data.success(collaborationItem);
+        } else {
+            data.failure(R.string.box_sharesdk_network_error);
+        }
+        return data;
+    }
+    /**
+     * Helper method for transforming BoxResponse to UI Model for getInviteesApi
+     * @param response the response to transform
+     * @return the transformed model
+     */
+    private DataWrapper<BoxIteratorInvitees> createGetInviteesItemData(BoxResponse<BoxIteratorInvitees> response) {
+        final DataWrapper<BoxIteratorInvitees> data = new DataWrapper<BoxIteratorInvitees>();
+        if (response.isSuccess()) {
+            final BoxIteratorInvitees invitees = response.getResult();
+            data.success(invitees);
+        } else {
+            BoxException boxException = (BoxException) response.getException();
+            int responseCode = boxException.getResponseCode();
+            int errorStrCode = R.string.box_sharesdk_generic_error;
+            if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
+                errorStrCode = R.string.box_sharesdk_insufficient_permissions;
+            } else if (boxException.getErrorType() == BoxException.ErrorType.NETWORK_ERROR) {
+                errorStrCode = R.string.box_sharesdk_network_error;
+            }
+            data.failure(errorStrCode);
+        }
+        return data;
+    }
+    /**
+     * Helper method for transforming BoxResponse to UI Model for addCollabsApi
+     * @param response the response to transform
+     * @return the transformed model
+     */
+    private InviteCollaboratorsDataWrapper createAddCollabsItemData(BoxResponse<BoxResponseBatch> response) {
+        return handleCollaboratorsInvited(response.getResult());
+    }
+
+    /**
+     * Helper method for helping with transformation of BoxBatchResponse to UI Model
+     * @param responses the batch response to transform
+     * @return the transformed model
+     */
+    @VisibleForTesting
+    private InviteCollaboratorsDataWrapper handleCollaboratorsInvited(BoxResponseBatch responses) {
+        int strCode = R.string.box_sharesdk_generic_error; //default generic error
+        boolean mInvitationFailed;
+        String subMssg;
+
+        int alreadyAddedCount = 0;
+        boolean didRequestFail = false;
+        String name = "";
+        HashSet<Integer> failureCodes = generateFailureCodes();
+        List<String> failedCollaboratorsList = new ArrayList<>();
+
+
+        for (BoxResponse<BoxCollaboration> r : responses.getResponses()) {
+            if (!r.isSuccess()) {
+                if (checkIfKnownFailure(r, failureCodes)) {
```

Comment:
`checkIfKnownFailure` returns true for HTTP_FORBIDDEN too - was this a new requirement? Earlier we only processed HTTP_BAD_REQUEST as a valid response code for invite failure due to collaborators error

Reply:
I saw it in the original code already.